controllers/dspipeline_controller.go

```diff
@@ -258,6 +258,8 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	dspaPrereqsReady := dbAvailable && objStoreAvailable
 
 	if dspaPrereqsReady {
+		params.SetupOwner(dspa)
+
```

Comment:
I don't see a reason why this needs to be a separate call to params, we generate all params at once so that subsequent calls that leverage params can make assume they have the same fields available to them, 

so I think it's better if we just call this logic within: 

```
err = params.ExtractParams(ctx, dspa, r.Client, r.Log)
```

Reply:
Makes sense. Done.